src/OrbitGl/App.cpp

```diff
@@ -999,65 +999,73 @@ void OrbitApp::SendErrorToUi(const std::string& title, const std::string& text)
-            modules_currently_loading_.erase(module_data->file_path());
-          });
-          return;
-        }
+      "Searching for symbols on remote instance (module \"%s\")...", module_file_path));
+
+  auto future = thread_pool_->Schedule([this, module_file_path]() -> ErrorMessageOr<std::string> {
+    return GetProcessManager()->FindDebugInfoFile(module_file_path);
+  });
+
+  return future.Then(
+      main_thread_executor_,
+      [this, module_file_path, scoped_status = std::move(scoped_status)](
+          ErrorMessageOr<std::string> result) mutable -> ErrorMessageOr<std::filesystem::path> {
+        if (!result) return result.error();
 
         const std::string& debug_file_path = result.value();
 
         LOG("Found symbols file on the remote: \"%s\" - loading it using scp...", debug_file_path);
 
-        main_thread_executor_->Schedule(
-            [this, module_data, function_hashes_to_hook = std::move(function_hashes_to_hook),
-             frame_track_function_hashes = std::move(frame_track_function_hashes), debug_file_path,
-             scoped_status = std::move(scoped_status)]() mutable {
-              const std::filesystem::path local_debug_file_path =
-                  symbol_helper_.GenerateCachedFileName(module_data->file_path());
-
-              {
-                scoped_status.UpdateMessage(
-                    absl::StrFormat(R"(Copying debug info file for "%s" from remote: "%s"...)",
-                                    module_data->file_path(), debug_file_path));
-                SCOPED_TIMED_LOG("Copying \"%s\"", debug_file_path);
-                auto scp_result =
-                    secure_copy_callback_(debug_file_path, local_debug_file_path.string());
-                if (!scp_result) {
-                  SendErrorToUi(
-                      "Error loading symbols",
-                      absl::StrFormat("Could not copy debug info file from the remote: %s",
-                                      scp_result.error().message()));
-                  modules_currently_loading_.erase(module_data->file_path());
-                  return;
-                }
-              }
-
-              LoadSymbols(local_debug_file_path, module_data, std::move(function_hashes_to_hook),
-                          std::move(frame_track_function_hashes));
-            });
+        const std::filesystem::path local_debug_file_path =
+            symbol_helper_.GenerateCachedFileName(module_file_path);
+
+        scoped_status.UpdateMessage(
+            absl::StrFormat(R"(Copying debug info file for "%s" from remote: "%s"...)",
+                            module_file_path, debug_file_path));
+        SCOPED_TIMED_LOG("Copying \"%s\"", debug_file_path);
+        auto scp_result = secure_copy_callback_(debug_file_path, local_debug_file_path.string());
+        if (!scp_result) {
+          return ErrorMessage{absl::StrFormat("Could not copy debug info file from the remote: %s",
```

Comment:
The message above says "from remote" while here we use the form "from the remote".